actions/setup.go

```diff
@@ -23,24 +24,77 @@ const (
 	KeepEnvironmentsAlways = "always"
 )
 
-// DefaultSuiteSetup holds the data for a default setup
-type DefaultSuiteSetup struct {
-	Config   *config.Config
+// NetworkInfo helps delineate network information in a multi-network setup
+type NetworkInfo struct {
 	Client   client.BlockchainClient
 	Wallets  client.BlockchainWallets
 	Deployer contracts.ContractDeployer
 	Link     contracts.LinkToken
-	Env      environment.Environment
 	Network  client.BlockchainNetwork
 }
 
-// DefaultLocalSetup setup minimum required components for test
-func DefaultLocalSetup(
-	envName string,
+// buildNetworkInfo initializes the network's blockchain client and gathers all test-relevant network information
+func buildNetworkInfo(network client.BlockchainNetwork, env environment.Environment) (NetworkInfo, error) {
+	// Initialize blockchain client
+	var bcc client.BlockchainClient
+	var err error
+	switch network.Config().Type {
+	case client.BlockchainTypeEVMMultinode:
+		bcc, err = environment.NewBlockchainClients(env, network)
+	case client.BlockchainTypeEVM:
+		bcc, err = environment.NewBlockchainClient(env, network)
+	}
+	if err != nil {
+		return NetworkInfo{}, err
+	}
+
+	// Initialize wallets
+	wallets, err := network.Wallets()
+	if err != nil {
+		return NetworkInfo{}, err
+	}
+	contractDeployer, err := contracts.NewContractDeployer(bcc)
+	if err != nil {
+		return NetworkInfo{}, err
+	}
+	link, err := contractDeployer.DeployLinkTokenContract(wallets.Default())
+	if err != nil {
+		return NetworkInfo{}, err
+	}
+	return NetworkInfo{
+		Client:   bcc,
+		Wallets:  wallets,
+		Deployer: contractDeployer,
+		Link:     link,
+		Network:  network,
+	}, nil
+}
+
+// SuiteSetup enables common use cases, and safe handling of different blockchain networks for test scenarios
+type SuiteSetup interface {
+	Config() *config.Config
+	Environment() environment.Environment
+
+	DefaultNetwork() NetworkInfo
+	Network(networkID string) (NetworkInfo, error)
```

Comment:
I'd change this function so it works by index, the same as we do with wallets. `Network(index int)`